tools/packaging/kernel/build-kernel.sh

```diff
@@ -247,6 +248,12 @@ get_kernel_frag_path() {
 		all_configs="${all_configs} ${conf_configs}"
 	fi
 
+	if [[ "${KATA_BUILD_CC}" == "yes" ]];then
+		info "Enabling config for confidential guest trust storage protection"
+		local cryptsetup_configs="$(ls ${common_path}/confidential_containers/cryptsetup.conf)"
+		all_configs="${all_configs} ${cryptsetup_configs}"
+	fi
```

Comment:
@arronwy, I would not take this path.

As we want this to be common to both TDX and SEV, I'd not put this under KATA_BUILD_CC, but rather in the `if` above, where we check for `conf_guest`.

By doing this we know for sure we have it enabled in both TDX and SEV cases, without the need to pass yet another environment variable.

```sh
diff --git a/tools/packaging/kernel/build-kernel.sh b/tools/packaging/kernel/build-kernel.sh
index 7090e28e5..6caddc5c1 100755
--- a/tools/packaging/kernel/build-kernel.sh
+++ b/tools/packaging/kernel/build-kernel.sh
@@ -243,15 +243,13 @@ get_kernel_frag_path() {
        fi
 
        if [[ "${conf_guest}" != "" ]];then
-               info "Enabling config for '${conf_guest}' confidential guest protection"
-               local conf_configs="$(ls ${arch_path}/${conf_guest}/*.conf)"
-               all_configs="${all_configs} ${conf_configs}"
-       fi
-
-       if [[ "${KATA_BUILD_CC}" == "yes" ]];then
                info "Enabling config for confidential guest trust storage protection"
                local cryptsetup_configs="$(ls ${common_path}/confidential_containers/cryptsetup.conf)"
                all_configs="${all_configs} ${cryptsetup_configs}"
+
+               info "Enabling config for '${conf_guest}' confidential guest protection"
+               local conf_configs="$(ls ${arch_path}/${conf_guest}/*.conf)"
+               all_configs="${all_configs} ${conf_configs}"
        fi
 
        info "Constructing config from fragments: ${config_path}"

``

Reply:
That's a good idea @fidencio. If we could avoid yet another envvar to control the script flow, the better.

Reply:
Thanks @fidencio fixed now :-)